includes/core-changes/corefx/3.0/disambiguate-generic-type-for-groupcollection.md

```diff
@@ -0,0 +1,67 @@
+### Passing GroupCollection to extension methods taking IEnumerable\<T> requires disambiguation
+
+When calling an extension method that takes an `IEnumerable<T>` on a <xref:System.Text.RegularExpressions.GroupCollection>, you must disambiguate the type using a cast.
+
+#### Change description
+
+Starting in .NET Core 3.0, <xref:System.Text.RegularExpressions.GroupCollection?displayProperty=nameWithType> implements `IEnumerable<KeyValuePair<String,Group>>` in addition to the other types it implements, including `IEnumerable<Group>`. This results in ambiguity when calling an extension method that takes an <xref:System.Collections.Generic.IEnumerable%601>. If you call such an extension method on a <xref:System.Text.RegularExpressions.GroupCollection> instance, for example, <xref:System.Linq.Enumerable.Count%2A?displayProperty=nameWithType>, you'll see the following compiler error:
+
+**CS1061: 'GroupCollection' does not contain a definition for 'Count' and no accessible extension method 'Count' accepting a first argument of type 'GroupCollection' could be found (are you missing a using directive or an assembly reference?)**
+
+In previous versions of .NET, there was no ambiguity and no compiler error.
+
+#### Version introduced
+
+3.0
+
+#### Reason for change
+
+This was an [unintentional breaking change](https://github.com/dotnet/corefx/pull/30077). Because it has been like this for some time, we don't plan to revert it. In additional, such a change would itself be breaking.
```

Comment:
```suggestion
This was an [unintentional breaking change](https://github.com/dotnet/corefx/pull/30077). Because it has been like this for some time, we don't plan to revert it. In addition, such a change would itself be breaking.
```